mailbox/backup/src/main/java/org/apache/james/mailbox/backup/DefaultMailboxBackup.java

```diff
@@ -81,12 +97,71 @@ public void backupAccount(User user, OutputStream destination) throws IOExceptio
         archive(mailboxes, messages, destination);
     }
 
-    private final MailboxManager mailboxManager;
-    private final ArchiveService archiveService;
+    @Override
+    public Publisher<Void> restore(User user, InputStream source) {
+        return Mono.fromRunnable(Throwing.runnable(() -> doRestore(user, source)).sneakyThrow())
+            .subscribeOn(Schedulers.elastic())
+            .doOnTerminate(Throwing.runnable(source::close))
+            .then();
+    }
+
+    private void doRestore(User user, InputStream source) throws MailboxException, IOException {
+        MailboxSession session = mailboxManager.createSystemSession(user.asString());
+        Map<String, MessageManager> oldToNewMailbox = new HashMap<>();
+        restoreEntries(source, session, oldToNewMailbox);
+    }
+
+    private void restoreEntries(InputStream source, MailboxSession session, Map<String, MessageManager> oldToNewMailbox) throws IOException {
+        try (MailArchiveIterator archiveIterator = archiveLoader.load(source)) {
+            archiveIterator.forEachRemaining(Throwing.consumer(entry -> {
+                restoreEntryWithErrorHandling(session, oldToNewMailbox, entry);
+            }));
+        }
+    }
+
+    private void restoreEntryWithErrorHandling(MailboxSession session, Map<String, MessageManager> oldToNewMailbox, MailArchiveEntry entry) throws MailboxException {
```

Comment:
OK to not enforce strong typing of origin mailbox ID

This might prevent us from using import-export across James back-end, what could be really useful

However I struggled understanding what that `String` actually represents...